libraries/AP_HAL_ChibiOS/hwdef/MFT-SEMA100/hwdef.dat

```diff
@@ -0,0 +1,223 @@
+PD12 I2C4_SCL I2C4
+PD13 I2C4_SDA I2C4
+
+# order of I2C buses
+I2C_ORDER I2C2 I2C3 I2C4 I2C1
+
+
+define HAL_I2C_INTERNAL_MASK 1
+
+# enable pins
+#PE3 VDD_3V3_SENSORS_EN OUTPUT LOW
+
+# drdy pins
+PE9  DRDY1_LIS3MDL INPUT
+
+
+# UARTs
+#USART1 TX RX 1
+PA10 USART1_RX USART1 
+PA9 USART1_TX USART1 
+
+# USART2 is TX RX 2
+PD6 USART2_RX USART2 
+PD5 USART2_TX USART2 
+
+# USART3 is TX RX 3
+PD9 USART3_RX USART3 
+PD8 USART3_TX USART3 
+
+# UART5 is TX RX 4
+PD2 UART5_RX UART5 
+PC12 UART5_TX UART5 
+
+# UART7 is TX RX 5
+PE7 UART7_RX UART7 
+PE8 UART7_TX UART7 
+
+# SBUS, DSM port
+PE0 UART8_RX UART8
+define DEFAULT_SERIAL5_PROTOCOL SerialProtocol_RCIN
+
+
+
+# PWM AUX channels
+PB0 TIM1_CH2N TIM1 PWM(1) GPIO(101)
+PB1 TIM1_CH3N TIM1 PWM(2) GPIO(102)
+PC7 TIM8_CH2 TIM8 PWM(3) GPIO(103)
+PC8 TIM8_CH3 TIM8 PWM(4) GPIO(104)
+PA0 TIM5_CH1 TIM5 PWM(5) GPIO(105)
+PA1 TIM5_CH2 TIM5 PWM(6) GPIO(106)
+PA2 TIM5_CH3 TIM5 PWM(7) GPIO(107)
+PA3 TIM5_CH4 TIM5 PWM(8) GPIO(108)
+
+PD14 TIM4_CH3  TIM4  PWM(9)  GPIO(50)
+PD15 TIM4_CH4  TIM4  PWM(10) GPIO(51)
+PA6 TIM3_CH1  TIM3  PWM(11) GPIO(52)
+PA7 TIM3_CH2  TIM3  PWM(12) GPIO(53)
+
+
+# allow for 14 PWMs by default
+
+# PWM output for buzzer
+PB15 TIM12_CH2 TIM12 GPIO(77) ALARM
+
+
+# analog in
+PA5 BATT_VOLTAGE_SENS ADC1 SCALE(1)
+#PA5 BATT2_VOLTAGE_SENS ADC1 SCALE(1)
+PC5 BATT_CURRENT_SENS ADC2 SCALE(1)
+
+
+define HAL_BATT_MONITOR_DEFAULT 4
+
+
+define HAL_BATT_VOLT_PIN 19
+define HAL_BATT_VOLT_SCALE 10
+
+#define HAL_BATT2_VOLT_PIN 15 
+#define HAL_BATT2_VOLT_SCALE 9
+
+define HAL_BATT_CURR_PIN 8
+define HAL_BATT_CURR_SCALE 10
+
+
+
+# CAN bus
+PD0 CAN1_RX CAN1
+PD1 CAN1_TX CAN1
+
+# 2nd CAN bus. Cannot be used at same time as USB_HS
+PB5 CAN2_RX CAN2
+PB6 CAN2_TX CAN2
+
+# control for silent (no output) for CAN
+PD4 GPIO_CAN1_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(70)
+PD11 GPIO_CAN2_SILENT OUTPUT PUSHPULL SPEED_LOW LOW GPIO(71)
+
+# GPIOs
+
+PD10 HEATER_EN OUTPUT LOW GPIO(80)
+define HAL_HEATER_GPIO_PIN 80
+define HAL_HAVE_IMU_HEATER 1
+
+
+#BARO
+BARO BMP388 I2C:1:0x77
+
+# SPI devices
+SPIDEV bmi088_a       SPI4 DEVID1  BMI088_A_CS     MODE3  10*MHZ 10*MHZ
+SPIDEV bmi088_g       SPI4 DEVID2  BMI088_G_CS     MODE3  10*MHZ 10*MHZ
+SPIDEV sdcard         SPI1 DEVID1  SDCARD_CS       MODE0  400*KHZ 25*MHZ
+
+
+# IMU
+IMU BMI088 SPI:bmi088_a SPI:bmi088_g ROTATION_PITCH_180_YAW_270
+
+define HAL_DEFAULT_INS_FAST_SAMPLE 7
+
+# compasses
+define HAL_PROBE_EXTERNAL_I2C_COMPASSES
+COMPASS LIS3MDL I2C:ALL_INTERNAL:0x1C false ROTATION_NONE
+
+
+#LEDs
+PC13 LED_BOOTLOADER OUTPUT HIGH GPIO(0)
+PC6 LED_ACTIVITY OUTPUT LOW GPIO(1) # optional
+
+define HAL_GPIO_A_LED_PIN 0
+define HAL_GPIO_B_LED_PIN 1
+
+#STORAGE 
+STORAGE_FLASH_PAGE 14
+define HAL_STORAGE_SIZE 32768
+
+# enable FAT filesystem support (needs a microSD defined via SDMMC)
+define HAL_OS_FATFS_IO 1
+
+DMA_PRIORITY ADC* SPI4* TIM*UP* 
+DMA_NOSHARE  SPI1* SPI2* SPI4* TIM*UP*
+
+define DEFAULT_NTF_LED_TYPES 743
+define DEFAULT_SERIAL4_BAUD 38
```

Comment:
dcuplicate in params....and SERIAL4 is defaulted to GPS protocol and baudrate is ignored anyway...if you have specific protocol target for Serial 3 and 4 other than GPS, their current default then perhaps changing default baud rate might make sense....but not currently

Reply:
Weird thing that when I check for the serials from MP the default Serial value was 230 for both Serıal 3 and 4 thats added to fix it. I can remove it if it is not necessary to show the default Baud as 38

Reply:
see below